### DISCUSSION TITLE
Reset Bootstrap components ID in snapshots

### DISCUSSION THREAD

**Discussion by forchel** [cdmoro/bootstrap-vue-3#977] [Q&A]
Hello, I started evaluating the bootstrap-vue-next package. Within vitests, we make extensive use of snapshots (inline or not) to test that our components render the correct markup. I noticed that the BVID changes with each execution. However, we need a fixed ID for each run. How can we best deal with this problem? Is there a way to reset the generated BVIDs for each test run?
Thanks.

PS: The Snapshot looks like:

```
// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginView > should render 1`] = `
<div data-v-387db8af="" class="card login-card">
  <!---->
  <!---->
  <div data-v-387db8af="" class="row g-0 p-3">
    <div data-v-387db8af="" class="col">
      <h4 data-v-387db8af="" class="card-title medium-title" title-tag="div">loginWord</h4>
      <form data-v-387db8af="" class="login-form">
        <fieldset data-v-387db8af="" class="mb-3" id="__BVID__164438___BV___">
          <!---->
          <div class=""><label data-v-ec7d3db4="" class="form-label">emailWord*</label><input data-v-ec7d3db4="" type="email" class="form-control" required="">
            <!--v-if-->
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
        <fieldset data-v-387db8af="" class="mb-3" id="__BVID__693252___BV___">
          <!---->
          <div class=""><label data-v-ec7d3db4="" class="form-label">loginPasswordWord*</label><input data-v-ec7d3db4="" type="password" class="form-control" required="">
            <!--v-if-->
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
        <div data-v-387db8af="" class="buttons mt-3"><button data-v-387db8af="" class="btn btn-primary btn-md disabled submit-button" disabled="" type="submit">
            <!---->
            <div class="btn-content">loginWord</div>
          </button><a data-v-387db8af="" class="btn-link btn">passwordForgottenWord</a></div>
      </form>
    </div>
  </div>
  <!---->
  <!---->
</div>
`;
```

**Comment by VividLemon** [comment]
![image](https://user-images.githubusercontent.com/31498096/227561693-1e140c1d-ca7a-403c-a144-911ce83482f9.png)

![image](https://user-images.githubusercontent.com/31498096/227561977-f0b9d912-8b83-417e-a9b0-12fe4d69cdac.png)

Any instance of an id being "dynamic" can be sidestepped by supplying your own id. If it doesn't, than it's a bug

**Reply by forchel** [reply]
Okay, then I'll remove all snapshots and test the DOM instead. Thx

**Reply by VividLemon** [reply]
It's generally a better idea to make your tests much more specific. You typically don't want to make tests brittle. Say, if you make one small change to a class, then everything breaks down quickly. Making tests more specific means you have much greater test coverage and prevention of issues. Snapshot tests are usually very weak and only tell you if something changed, not necessarily how.
